### DISCUSSION TITLE
バージョンごとのgit tagの復活

### DISCUSSION THREAD

**Discussion by Shimi9999** [exch-bms2/beatoraja#670] [Ideas]
最新バージョンのリリースがMocha-Repository上に統一されて以来(0.7.7以降)、バージョンのタグ付けは行われていません。
GitHub上でのリリースは不要ですが、タグは付けるようにしてほしいです。

タグ付けのメリット
- どの機能がどのバージョンで入ったかが判別しやすくなる
- 最新バージョンでどんな機能が入ったかが分かりやすくなり、最新バージョンの導入促進に繋がる(これが大きいと思う)
- バグやデグレが発生した時に、影響バージョンの範囲が調べやすくなる

タグ付けのデメリット
- リリース者(@exch-bms2)の作業量が増える

**Comment by exch-bms2** [comment]
正直バージョン番号についてはあまり意味がないと思っているので、リリースノートをプロジェクトに含めるようにしましょうか。
機能追加、バグ修正時にリリースノートも更新してコミットする感じで。
上記3点は解決すると思います。

**Comment by Shimi9999** [comment]
各コントリビューターにリリースノート更新を任せるのはややコストが高いと思うので、以下のようなリリースノート自動生成機能を使うのはどうでしょう。

https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

この方法だと結局GitHubのリリースとタグの機能を使うことにはなりますが…。
あと項目をプルリクから作成するので、masterに直接pushされた追加修正は加筆する必要がありそうです。

結局一番低コストなのはタグ付けな気もします。

**Comment by exch-bms2** [comment]
それだとユーザーに開示する必要のないコミット(内部リファクタ等)も含まれるので情報過多になってしまい、
情報取捨選択に余計なコストがかかる懸念があると思います。
主要な更新に気づいた人がリリースノート更新でいいかな、と。

**Comment by Shimi9999** [comment]
現状"そのバージョンで含まれているコミットはどこからどこまで"という情報はリリース者以外は分からないので、タグやコミットハッシュなどでバージョンの区切りを示す必要があると思います。
あと、プロジェクト内にリリースノートを含むと書き換え時に毎回プルリクを出すのが手間になりそうなので、やるならWikiに変更をまとめていくのが良いと思うのですがどうでしょうか。
